### PR TITLE
Breakout development services & improve naming

### PR DESCRIPTION
**Pull request by iamcarbon**
- Moves the DevelopmentInMemoryStore, StoredCredential, and ConformanceMetadataService into a new Fido2.Development project.

- Renames a few properties on StoredCredential for clarity.
- Removes duplicate properties on StoredCredential (preferring specification name)

### CODE REVIEW

File: BlazorWasmDemo/Client/BlazorWasmDemo.Client.csproj [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <TargetFramework>net6.0</TargetFramework>
     <Nullable>enable</Nullable>
     <ImplicitUsings>enable</ImplicitUsings>
+    <LangVersion>11</LangVersion>
```
Comment (Regenhardt) [inline BlazorWasmDemo/Client/BlazorWasmDemo.Client.csproj:7]: Better remove this property from all the projects and update the one in the global Directory.Build.props instead.

File: Src/Fido2.Development/StoredCredential.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    public List<byte[]> DevicePublicKeys { get; set; }
+
+    public byte[] UserId { get; set; }
+
+    public PublicKeyCredentialDescriptor Descriptor { get; set; }
+
+    public byte[] UserHandle { get; set; }
+
+    public string CredType { get; set; }
```
Comment (Regenhardt) [inline Src/Fido2.Development/StoredCredential.cs:59]: Could rename this to `AttestationFormat`, see #426.

**Comment by Regenhardt** [comment]
This will definitely complete 3) from #426, nice.

File: Src/Fido2.Development/StoredCredential.cs (hunk 2 above)
Comment (iamcarbon) [reply Src/Fido2.Development/StoredCredential.cs]: Agreed that AttestationFormat is a better name here. Updated.

**Comment by iamcarbon** [comment]
@abergs Ready for review. This partially addresses the feedback received in #427

**Comment by abergs** [comment]
Thanks @iamcarbon and @Regenhardt - I've done a first pass, but will need more time to review this as it's extensive changes. Combined with a busy schedule, it make take some extra time before I'm able to merge. Thanks for the contributions and patience.

**Comment by abergs** [comment]
Just an update, I will start work on this.

**Review by jonashendrickx — APPROVED** [review]

**Comment by iamcarbon** [comment]
@abergs I think we have two options on CredentialId. 

1. Drop it from AttestationVerificationSuccess (it was never set)
2. Rename Id to CredentialId. 

My thinking on keeping Id (and dropping CredentialId) is explained in #428. The meaning of Id could also be clarified renaming the type to something ending in Credential.

File: BlazorWasmDemo/Client/BlazorWasmDemo.Client.csproj (hunk 1 above)
Comment (abergs) [reply BlazorWasmDemo/Client/BlazorWasmDemo.Client.csproj:7]: @iamcarbon Agreed, but could be separate PR.

File: Src/Fido2/Objects/AuthenticatorData.cs [hunk 3]
```diff
@@ -79,7 +79,7 @@ public AuthenticatorData(byte[] rpIdHash, AuthenticatorFlags flags, uint signCou
     /// Backup state is signaled in authenticator data's flags and can change over time.
     /// <see cref="https://w3c.github.io/webauthn/#backup-state"/>
     /// </summary>
-    public bool BackupState => _flags.HasFlag(AuthenticatorFlags.BS);
+    public bool IsBackedUp => _flags.HasFlag(AuthenticatorFlags.BS);
```
Comment (abergs) [inline Src/Fido2/Objects/AuthenticatorData.cs:82]: I was contemplating if we should always keep using Spec terms on places you added `IsBackedUp`.

I decided the readability is worth not aligning to spec (as long as we mention BS/BE in XML).

**Review by abergs — APPROVED** [review]

**Comment by abergs** [comment]
@iamcarbon While this PR is accepted as is - It would be nice to change the name of `AttestationVerificationSuccess` to `RegisteredPublicKeyCredential ` - it would make DX a bit better I think.

**Comment by iamcarbon** [comment]
@abergs All set.

**Pull request merged**